//Define the used version of proto
syntax = "proto3";

package pomdpir;

//The first step is to solve the model and compute a policy
message ComputePolicyRequest {
  int32 nRounds = 1 ;
  int32 numBelStates = 2 ;
  int32 maxBelStates = 3;
  int32 episodeLength = 4 ;
  float threshold = 5 ;
  float explorProb = 6 ;
  int32 nIterations = 7 ;
  int32 maxAlphaSetSize = 8 ;

}

//Finally, at execution time the user wants to query the policy to obtain an action. For that he must send an observation
message StepRequest {
  repeated int32 observations = 1 ;
}

message StepResponse {
  int32 action = 1 ;
  string action_name = 2 ;
}

message Empty {
}

//Define the service
service POMDPIR {
  rpc compute_policy(ComputePolicyRequest) returns (Empty);
  rpc reset_execution(Empty) returns(Empty);
  rpc step(StepRequest) returns(StepResponse);
}